pkg/steps/cluster_claim.go

```diff
@@ -122,23 +124,12 @@ func acquireCluster(ctx context.Context, clusterClaim api.ClusterClaim, hiveClie
 		return nil, fmt.Errorf("failed to created cluster claim %s in namespace %s: %w", claim.Name, claim.Namespace, err)
 	}
 	logrus.Info("Waiting for the claimed cluster to be ready.")
+	if err := waitForClaim(hiveClient, claimNamespace, claimName, clusterClaim.Timeout.Duration); err != nil {
+		return nil, fmt.Errorf("failed to wait for created cluster claim to become ready: %w", err)
+	}
 	claim = &hivev1.ClusterClaim{}
-	if err := wait.Poll(15*time.Second, clusterClaim.Timeout.Duration, func() (bool, error) {
-		if err := hiveClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: claimName, Namespace: claimNamespace}, claim); err != nil {
-			return false, fmt.Errorf("failed to get cluster claim %s in namespace %s: %w", claim.Name, claim.Namespace, err)
-		}
-		// https://github.com/openshift/hive/blob/a535d29c4d2dc4f7f9bf3f30098c69d9334bee2e/apis/hive/v1/clusterclaim_types.go#L18-L22
-		for _, condition := range claim.Status.Conditions {
-			logrus.WithField("claim.Name", claim.Name).WithField("claim.Namespace", claim.Namespace).
-				WithField("condition.Type", condition.Type).WithField("condition.Status", condition.Status).
-				Debug("Checking the claim's status.")
-			if condition.Type == hivev1.ClusterRunningCondition && condition.Status == corev1.ConditionTrue {
-				return true, nil
-			}
-		}
-		return false, nil
-	}); err != nil {
-		return nil, fmt.Errorf("failed to wait for created cluster claim to become running: %w", err)
+	if err := hiveClient.Get(ctx, ctrlruntimeclient.ObjectKey{Name: claimName, Namespace: claimNamespace}, claim); err != nil {
```

Comment:
The `into` is set by the generic function. You do not need this.

Reply:
Updated.